firebase_test.go

```diff
@@ -52,6 +53,44 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestNewFromHttpClient(t *testing.T) {
+	t.Parallel()
+
+	var tr *http.Transport
+	tr = &http.Transport{
+		DisableKeepAlives: true, // https://code.google.com/p/go/issues/detail?id=3514
+		Dial: func(network, address string) (net.Conn, error) {
+			start := time.Now()
+			c, err := net.DialTimeout(network, address, TimeoutDuration)
+			tr.ResponseHeaderTimeout = TimeoutDuration - time.Since(start)
+			return c, err
+		},
+	}
+	var client = http.Client{Transport: tr};
+
+	testCases := []struct {
+		givenURL string
```

Comment:
Agreed @hostirosti . I think the test is find as is given that it's already in the source.